pvlib/shading.py

```diff
@@ -191,3 +192,63 @@ def sky_diffuse_passias(masking_angle):
        Available at https://www.nrel.gov/docs/fy18osti/67399.pdf
     """
     return 1 - cosd(masking_angle/2)**2
+
+
+def shaded_fraction(solar_zenith, solar_azimuth, surface_tilt, surface_azimuth,
+                    gcr, cross_axis_tilt=0.):
+    """
+    Calculate fraction (from the bottom) of row slant height that is shaded
+    by the row in front toward the sun.
+
+    See [1], Eq. 32.
+
+    .. math::
+        F_x = \\max \\left( 0, \\min \\left(\\frac{\\text{GCR} \\cos \\theta
+        + \\left( \\text{GCR} \\sin \\theta - \\tan \\beta_{c} \\right)
+        \\tan Z - 1}
+        {\\text{GCR} \\left( \\cos \\theta + \\sin \\theta \\tan Z \\right)},
+        1 \\right) \\right)
+
+    Parameters
+    ----------
+    solar_zenith : numeric
+        Apparent solar zenith. [degrees]
+    solar_azimuth : numeric
+        Solar azimuth. [degrees]
+    surface_tilt : numeric
```

Comment:
Need to re-test this, according to the reference this should be rotation, not tilt.

Reply:
Another consideration is that the solar projection must always be a rotation around the same axis as the surface rotation (no 180 degree axis azimuth flip).  I'm not sure that's possible with with current definition of `bifacial.utils._solar_projection_tangent`, but I didn't look very closely so I might be wrong.